doc/remediation-templating.md

```diff
@@ -0,0 +1,352 @@
+# Remediation template proposed flow design
+This page describes the design and implementation of the remediation templating
+support in the compliance operator.
+
+## Goals
+We want to accomplish:
+   * The remediation content can use variables that SCAP supports instead of a pre-set 
+     fixed value
+   * During a scan where an administrator set custom variables in the tailored profile, it
+     is possible to create `ComplianceRemediation` objects based on the given custom values
+   * As a content writer, it is possible to use XCCDF value from rules in Kubernetes remediation
+   * Add additional remediation state that shows "Needs-Review" for not set and no default values 
+     remediations
+
+## High-level overview
+
+The compliance operator allows an administrator to extend a pre-existing profile, 
+to enable, disable rules and set values that come from the ProfileBundle. When 
+this happens, a CR `TailoredProfile` will be generated. Currently the `ComplianceScan`
+will use the tailored rules and variables from `TailoredProfile` CR, however,
+the `ComplianceRemediation` will now take custom values from `TailoredProfile`.
+Here we are going to implement that feature.
+
+### ComplianceAsCode remediation rule format
+
+
+In ComplianceAsCode, when the SCAP remediation content is generated, normally we will 
+have fixed remediation. In this proposal, we are going to change the remediation rule
+formate for the Kubernetes to include a marked reference variable to reference related
+XCCDF values from rules. 
+
+The following example is parts of two YAML files from ComplianceAsCode content, 
+the first one is the rule, and the second is the remediation:
+```yaml
+documentation_complete: true
+title: 'Set SSH Idle Timeout Interval'
+...
+requires:
+    - sshd_set_keepalive
+
+ocil_clause: 'it is commented out or not configured properly'
+
+ocil: |-
+    Run the following command to see what the timeout interval is:
+    <pre>$ sudo grep ClientAliveInterval /etc/ssh/sshd_config</pre>
+    If properly configured, the output should be:
+    <pre>ClientAliveInterval {{{ xccdf_value("sshd_idle_timeout_value") }}}</pre>
+...
+```
+As showing in the rule YAML file above, an XCCDF value for SSH Idle Timeout Interval is set.
+
+In the remediation YAML file, we can modify that Marco to pass the name of the 
```

Comment:
nit: s/Marco/macro/

Reply:
Thanks for the correction, it is now ready for the review! I think we can define `missing_value = []string{}`, and add all the name of missing value variables here, and then to check if this missing_value is empty, if not we set label to the `missing_value` so that we know all the missing variables